pykickstart/commands/ntp.py

```diff
@@ -0,0 +1,137 @@
+# will be useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  Any Red Hat
+# trademarks that are incorporated in the source code or documentation are not
+# subject to the GNU General Public License and may only be used or replicated
+# with the express permission of Red Hat, Inc.
+#
+from pykickstart.errors import KickstartParseError
+from pykickstart.version import F33
+from pykickstart.base import BaseData, KickstartCommand
+from pykickstart.options import KSOptionParser
+
+from pykickstart.i18n import _
+
+class F33_NTPData(BaseData):
+    removedKeywords = BaseData.removedKeywords
+    removedAttrs = BaseData.removedAttrs
+
+    def __init__(self, *args, **kwargs):
+        BaseData.__init__(self, *args, **kwargs)
+        self.server = kwargs.get("server", [])
+        self.pool = kwargs.get("pool", "")
+        self.nts = kwargs.get("nts", False)
+
+    def __eq__(self, y):
+        if not y:
+            return False
+
+        if self.server:
+            return self.server == y.server and self.nts == y.nts
+        elif self.pool:
+            return self.pool == y.pool and self.nts == y.nts
+
+    def __ne__(self, y):
+        return not self == y
+
+    def __str__(self):
+        retval = BaseData.__str__(self)
+
+        args = self._getArgsAsStr()
+        if args:
+            retval += "ntp%s\n" % args
+
+        return retval
+
+    def _getArgsAsStr(self):
+        retval = ""
+
+        if self.server:
+            retval += " --server=%s" % self.server
+        if self.pool:
+            retval += " --pool=%s" % self.pool
+        if self.nts:
+            retval += " --nts"
+
+        return retval
+
+
+class F33_NTP(KickstartCommand):
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+
+    def __init__(self, writePriority=0, *args, **kwargs):
+        KickstartCommand.__init__(self, writePriority, *args, **kwargs)
+        self.op = self._getParser()
+        self.NTPList = kwargs.get("NTPList", [])
+        self.exclusive_required_options = [("server", "--server"),
+                                           ("pool", "--pool")]
+
+
+    def __str__(self):
+        retval = ""
+        for ntp_command in self.NTPList:
+            retval += ntp_command.__str__()
+
+        return retval
+
+    def _getParser(self):
+        op = KSOptionParser(prog="ntp", description="""
+                            Configures the Network Time Protocol.""",
+                            version=F33)
+        op.add_argument("--server", version=F33, help="""
```

Comment:
We should also support the `--disabled` option.

Reply:
Added as `--ntp-disable`. :)